src/python/pants/notes/2.7.x.md

```diff
@@ -1,3 +1,43 @@
+# 2.7.x Stable Releases
+
+## 2.7.0.dev3 (Aug 06, 2021)
+
+### New Features
+
+* Add `skip_tests` field to `python_tests` to facilitate incremental migrations ([#12510](https://github.com/pantsbuild/pants/pull/12510))
+
+* Add experimental lockfile support to IPython, Pytest, and Pylint ([#12491](https://github.com/pantsbuild/pants/pull/12491))
+
+* Adds support for Poetry group dependencies ([#12492](https://github.com/pantsbuild/pants/pull/12492))
+
+* Tailor pex_binary targets for entry points. ([#12478](https://github.com/pantsbuild/pants/pull/12478))
+
+* Go: refactor Go SDK invocation into common rule ([#12494](https://github.com/pantsbuild/pants/pull/12494))
+
+* Go: dependency inference for packages ([#12484](https://github.com/pantsbuild/pants/pull/12484))
```

Comment:
I think we've been putting these as internal because it's not yet usable enough for end users. These are notes for end users, not for us. For us, we have `git log`.